### PR TITLE
JBEHAVE-1471 Assign unique IDs for Story and Scenario entities

### PR DESCRIPTION
**Pull request by uarlouski**

### CODE REVIEW

File: jbehave-core/src/main/java/org/jbehave/core/model/Scenario.java [hunk 1]
```diff
@@ -69,6 +71,10 @@ public ExamplesTable getExamplesTable() {
         return examplesTable;
     }
 
+    public String getUuid() {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/model/Scenario.java]: `getId()`?

File: jbehave-core/src/main/java/org/jbehave/core/model/Story.java [hunk 2]
```diff
@@ -151,6 +153,10 @@ public String getName() {
         return (name != null ? name : getPath());
     }
 
+    public String getUuid() {
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/model/Story.java]: `getId()`?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: jbehave-core/src/main/java/org/jbehave/core/reporters/HtmlOutput.java [hunk 3]
```diff
@@ -50,7 +50,7 @@ public HtmlOutput(PrintStream output, Properties outputPatterns,
     private static Properties defaultHtmlPatterns() {
         Properties patterns = new Properties();
         patterns.setProperty("dryRun", "<div class=\"dryRun\">{0}</div>\n");        
-        patterns.setProperty("beforeStory", "<div class=\"story\">\n<h1>{0}</h1>\n<div class=\"path\">{1}</div>\n");
+        patterns.setProperty("beforeStory", "<div class=\"story\">\n<h1>{0}</h1>\n<div class=\"path\">{1}</div>\n<div class=\"id\">{2}</div>\n");
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/HtmlOutput.java]: `class` -> `id`

File: jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java [hunk 4]
```diff
@@ -99,7 +99,7 @@ protected String transformPrintingTable(String text, String tableStart, String t
     private static Properties defaultXmlPatterns() {
         Properties patterns = new Properties();
         patterns.setProperty("dryRun", "\"dryRun\": \"{0}\"");
-        patterns.setProperty("beforeStory", "'{'\"path\": \"{1}\", \"title\": \"{0}\"");
+        patterns.setProperty("beforeStory", "'{'\"path\": \"{1}\", \"title\": \"{0}\", \"id\": \"{2}\"");
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]: `id` should go first

File: jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java [hunk 5]
```diff
@@ -137,7 +137,7 @@ private static Properties defaultXmlPatterns() {
         patterns.setProperty("afterAfterScenarioSteps", "]");
         patterns.setProperty("beforeScenarioSteps", "\"steps\": [");
         patterns.setProperty("afterScenarioSteps", "]");
-        patterns.setProperty("beforeScenario","'{'\"keyword\": \"{0}\", \"title\": \"{1}\"");
+        patterns.setProperty("beforeScenario","'{'\"keyword\": \"{0}\", \"title\": \"{1}\", \"id\": \"{2}\"");
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/JsonOutput.java]: `id` should go first

File: jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java [hunk 6]
```diff
@@ -238,7 +238,8 @@ public void storyCancelled(Story story, StoryDuration storyDuration) {
 
     @Override
     public void beforeStory(Story story, boolean givenStory) {
-        print(format("beforeStory", "{0}\n({1})\n", story.getDescription().asString(), story.getPath()));
+        print(format("beforeStory", "{0}\n({1})\n({2})\n", story.getDescription().asString(), story.getPath(),
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java]: `id` should go first

File: jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java [hunk 7]
```diff
@@ -429,7 +430,7 @@ public void scenarioExcluded(Scenario scenario, String filter) {
     @Override
     public void beforeScenario(Scenario scenario) {
         cause.set(null);
-        print(format("beforeScenario", "{0} {1}\n", keywords.scenario(), scenario.getTitle()));
+        print(format("beforeScenario", "{0} {1} {2}\n", keywords.scenario(), scenario.getTitle(), scenario.getId()));
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java]: `id` should go first

Comment (valfirst) [reply jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java]: the same for other similar parts

**Review by valfirst — CHANGES_REQUESTED** [review]

File: jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java [hunk 8]
```diff
@@ -238,7 +238,8 @@ public void storyCancelled(Story story, StoryDuration storyDuration) {
 
     @Override
     public void beforeStory(Story story, boolean givenStory) {
-        print(format("beforeStory", "{0}\n({1})\n", story.getDescription().asString(), story.getPath()));
+        print(format("beforeStory", "{2}\n{0}\n({1})\n", story.getDescription().asString(), story.getPath(),
```
Comment (valfirst) [inline jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java]: why not to have `"{0}\n{1}\n({2})\n"` and adjust the sequence of arguments?

Comment (valfirst) [reply jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java]: the same for other similar places

**Review by valfirst — CHANGES_REQUESTED** [review]

File: jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java (hunk 8 above)
Comment (uarlouski) [reply jbehave-core/src/main/java/org/jbehave/core/reporters/PrintStreamOutput.java]: wanted to do it separately to show visibility in commits about reordering

**Review by valfirst — APPROVED** [review]

**Comment by uarlouski** [comment]
In master https://github.com/jbehave/jbehave-core/commit/e8597bdbdafd76ac15bad4562acccba62c463297

**Pull request closed without merging**